app/assets/javascripts/discourse/app/components/topic-timeline.js

```diff
@@ -56,45 +47,47 @@ export default MountWidget.extend(Docking, {
     this.queueRerender();
   },
 
-  dockCheck(info) {
-    const mainOffset = $("#main").offset();
-    const offsetTop = mainOffset ? mainOffset.top : 0;
-    const topicTop = $(".container.posts").offset().top - offsetTop;
-    const topicBottom =
-      $("#topic-bottom").offset().top - $("#main-outlet").offset().top;
+  dockCheck() {
+    const topicTop = domUtils.offset(document.querySelector(".container.posts"))
+      .top;
+    const topicBottom = domUtils.offset(document.querySelector("#topic-bottom"))
+      .top;
+
     const timeline = this.element.querySelector(".timeline-container");
     const timelineHeight = (timeline && timeline.offsetHeight) || 400;
-    const footerHeight = $(".timeline-footer-controls").outerHeight(true) || 0;
 
     const prev = this.dockAt;
-    const posTop = headerPadding() + info.offset();
-    const pos = posTop + timelineHeight - outletHeights();
+    const posTop = this.headerOffset() + window.pageYOffset;
+    const pos = posTop + timelineHeight;
 
     this.dockBottom = false;
     if (posTop < topicTop) {
       this.dockAt = parseInt(topicTop, 10);
-    } else if (pos > topicBottom + footerHeight) {
-      this.dockAt = parseInt(
-        topicBottom - timelineHeight + footerHeight + outletHeights(),
-        10
-      );
+    } else if (pos > topicBottom) {
+      this.dockAt = parseInt(topicBottom - timelineHeight, 10);
       this.dockBottom = true;
       if (this.dockAt < 0) {
         this.dockAt = 0;
       }
     } else {
       this.dockAt = null;
-      this.fastDockAt = parseInt(
-        topicBottom - timelineHeight + footerHeight - offsetTop,
-        10
-      );
+      this.fastDockAt = parseInt(topicBottom - timelineHeight, 10);
     }
 
     if (this.dockAt !== prev) {
       this.queueRerender();
     }
   },
 
+  headerOffset() {
+    return (
+      parseInt(
+        getComputedStyle(document.body).getPropertyValue("--header-offset"),
```

Comment:
`getComputedStyle` can be very expensive, because it can trigger a re-calculation of styles for the entire document.

Since we know that the `--header-offset` property will be set directly on the `<body>`, and will not be cascaded from some parent element, I think we could do something like:

```suggestion
        document.body.style.getPropertyValue("--header-offset"),
```